### ISSUE TITLE
Handle Spaces in Window Class Names

### ISSUE THREAD

**Issue by d-air1** [anametologin/krohnkite#78]
**Symptom**
When trying to add a window to the ignore list so that it doesn't get tiled, spaces seem to be ignored.
For example some apps such as qmmp, vorta, and basically anything with python have their window class as basically `python [app name]`. Heed the space.

**How to Reproduce**
1. Add a window to the ignore list by class in krohnkite settings.
2. Apply krohnkite settings and toggle krohnkite on and off in system settings.

**Expected behavior**
If an application or window has a space in the class name then the whole thing should be treated as the class name. Currently, it seems like adding any name that contains a space in the ignore by class field instead gets ignored itself. I don't want to match all apps written in python by adding python to the class role and I have also had issues with instead trying to match by window title if the window title is something generic such as the word `delete` or `configure`.

**Environment**
 - Distro: Arch
 - KWin version: 6.1.90 (Currently beta testing Plasma 6.2)
 - Krohnkite version: 0.9.8
 - List of KWin scripts in use: just this one.

**Notes**
The ignore by title section seems to support spaces just fine and even seems to support regex. It would be incredibly beneficial to have some of that in the ignore by class section as well. Unless I just missed how to do it and it is possible. In that case it would be nice if it was documented in the readme.

Lastly: Thanks for all your work.

**Comment by anametologin** [comment]
https://github.com/anametologin/krohnkite#search-a-window-parameters-to-filter-float-etc
qmmp for example has: `qml: krohnkite:<caption: Qmmp> <output.name: DP-2> <resourceName: qmmp> <resourceClass: qmmp>`

**Comment by d-air1** [comment]
> https://github.com/anametologin/krohnkite#search-a-window-parameters-to-filter-float-etc qmmp for example has: `qml: krohnkite:<caption: Qmmp> <output.name: DP-2> <resourceName: qmmp> <resourceClass: qmmp>`

I recall reading that section and that will give the resourceClass as `qmmp`. However, adding the word `qmmp` to my existing list of ignored windows seperated by commas does not actually work. Unless I'm doing that wrong.

What I have tried:
lower case `qmmp`
upper case `Qmmp`
and all caps `QMMP`

So my assumption was that it was using the whole class name. Filtering by title works in this case.

Using the kwin debug konsole, I noticed that qmmp is the only x11 window I have opened and the rest are wayland. Not sure if that is a problem just something to note and that I will test later on.

**Comment by d-air1** [comment]
Alright, so I seemed to have the solved the issue. For whatever reason, I decided to add both the lowercase and uppercase comma versions (`qmmp` and `Qmmp`) to my comma seperated list. Only now does it work. I have tested removing one or the other and then it goes back to not working. Only by having both of these together does it work. Specifically for this application.

I guess this means my assumptions of it using the whole application class which contained a space was wrong.
